test/stress.cxx

```diff
@@ -1604,6 +1607,37 @@ void stress16()
    if (gPrintSubBench) { printf("Test 16 : "); gBenchmark->Show("stress");gBenchmark->Start("stress"); }
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// Test mkdir returnExistingDirectory outside and within a Root file
+/// Create some directories, ensure they point to the expected places
+
+void stress17()
+{
+   Bprint(17,"Test mkdir");
+   
+   // check mkdir functions as expected in TDirectory
+   TDirectory *free_motherdir = new TDirectory("free_motherdir", "free_motherdir");
+   TDirectory *free_daughterdir = free_motherdir->mkdir("free_daughterdir");
+   TDirectory *free_daughter2 = free_motherdir->mkdir("free_daughterdir");
+   TDirectory *free_daughtersame = free_motherdir->mkdir("free_daughterdir", "", true);
+   
+   // check mkdir functions as expected inside a file
+   TFile f("stress.root","update");
+   TDirectory *motherdir = f.mkdir("motherdir");
+   TDirectory *daughterdir = motherdir->mkdir("daughterdir");
+   TDirectory *daughternull = motherdir->mkdir("daughterdir");
+   TDirectory *daughtersame = motherdir->mkdir("daughterdir", "", true);
+   
+   Bool_t OK = kTRUE;
+   if (daughternull != nullptr || daughtersame != daughterdir || free_daughter2 == free_daughterdir || free_daughtersame != free_daughterdir) OK = kFALSE;
+   if (OK) printf("OK\n");
+   else    {
+      printf("FAILED\n");
+      printf("%-8s free_daughterdir=%d, free_daughter2=%d, free_daughtersame=%d, daughterdir=%d, daughternull=%d, daughtersame=%d \n"," ",free_daughterdir,free_daughter2,free_daughtersame,daughterdir,daughternull,daughtersame);
+   }
+   if (gPrintSubBench) { printf("Test 17 : "); gBenchmark->Show("stress");gBenchmark->Start("stress"); }
+}
+
```

Comment:
test-stress now fails like 
```text
$ ctest -j8 -I "[0,136,136]" -V
UpdateCTestConfiguration  from :/Applications/root_build/DartConfiguration.tcl
Parse Config file:/Applications/root_build/DartConfiguration.tcl
 Add coverage exclude regular expressions.
SetCTestConfiguration:CMakeCommand:/Applications/CMake.app/Contents/bin/cmake
UpdateCTestConfiguration  from :/Applications/root_build/DartConfiguration.tcl
Parse Config file:/Applications/root_build/DartConfiguration.tcl
Test project /Applications/root_build
Constructing a list of tests
Ignore test: roottest-cling-parsing-semicolon
Done constructing a list of tests
Updating test list for fixtures
Added 0 tests to meet fixture requirements
Checking test dependency graph...
Checking test dependency graph end
test 136
    Start 136: test-stress

136: Test command: /Applications/CMake.app/Contents/bin/cmake "-DCMD=/Applications/root_build/test/stress^-b" "-DSYS=/Applications/root_build" "-P" "/Applications/root_build/RootTestDriver.cmake"
136: Environment variables: 
136:  ROOT_HIST=0
136: Test timeout computed to be: 1500
136: ******************************************************************
136: *  Starting  R O O T - S T R E S S test suite with 1000 events
136: ******************************************************************
136: Test  1 : Functions, Random Numbers, Histogram Fits............. OK
136: Test  2 : Check size & compression factor of a Root file........ OK
136: Test  3 : Purge, Reuse of gaps in TFile......................... OK
136: Test  4 : Test of 2-d histograms, functions, 2-d fits........... OK
136: Test  5 : Test graphics & Postscript............................ OK
136: Test  6 : Test subdirectories in a Root file.................... OK
136: Test  7 : TNtuple, selections, TCut, TCutG, TEventList.......... OK
136: Test  8 : Trees split and compression modes..................... OK
136: Test  9 : Analyze Event.root file of stress 8................... OK
136: Test 10 : Create 10 files starting from Event.root.............. OK
136: Test 11 : Test chains of Trees using the 10 files............... Error in <TDirectoryFile::mkdir>: An object with name daughterdir exists already
136: OK
136: Test 12 : Compare histograms of test 9 and 11................... OK
136: Test 13 : Test merging files of a chain......................... OK
136: Test 14 : Check correct rebuilt of Event.root in test 13........ OK
136: Test 15 : Divert Tree branches to separate files................ OK
136: Test 16 : CINT test (3 nested loops) with LHCb trigger.......... OK
136: Test 17 : Test mkdir............................................ OK
136: ******************************************************************
136: *  SYS: Darwin Michaels-MacBook-Pro.local 19.0.0 Darwin Kernel Versi
136: *  SYS: 10.15.1 Mac OS X 
136: ******************************************************************
136: stress    : Total I/O =  556.5 Mbytes, I =  408.2, O = 148.3
136: stress    : Compr I/O =  384.8 Mbytes, I =  280.2, O = 104.5
136: stress    : Real Time =  30.10 seconds Cpu Time =  19.66 seconds
136: ******************************************************************
136: *  ROOTMARKS =1888.5   *  Root6.19/01   20190529/1059
136: ******************************************************************
1/1 Test #136: test-stress ......................***Failed  Error regular expression found in output. Regex=[FAILED|Error in] 30.74 sec

0% tests passed, 1 tests failed out of 1

Label Time Summary:
longtest    =  30.74 sec*proc (1 test)

Total Test time (real) =  32.63 sec

The following tests FAILED:
	136 - test-stress (Failed)
Errors while running CTest
```
@pcanal Do you have any ideas what's causing this? I don't see anything in my added code that could be causing it.

Reply:
Technically, it complains because:
```
136: Test 11 : Test chains of Trees using the 10 files............... Error in <TDirectoryFile::mkdir>: An object with name daughterdir exists already
136: OK
```

Reply:
i.e. the 'stress' example is not allowed to issue any error message.

Reply:
Rather than extending test/stress is might be better to add an individual test under io/io/test

Reply:
Ah, I see. Thanks. I dropped the `daughternull` check, which, after all, is designed to produce an error, and it works fine now. I think this is a sufficient test, but if you disagree or strongly prefer it in the io/io/test directory, let me know.